pyvistaqt/rwi.py

```diff
@@ -0,0 +1,774 @@
+# modified from: https://gitlab.kitware.com/vtk/vtk/-/blob/master/Wrapping/Python/vtkmodules/qt/QVTKRenderWindowInteractor.py
+# under the OSI-approved BSD 3-clause License
+# pylint: skip-file
+# coding=utf-8
+"""
+A simple VTK widget for PyQt or PySide.
+See http://www.trolltech.com for Qt documentation,
+http://www.riverbankcomputing.co.uk for PyQt, and
+http://pyside.github.io for PySide.
+
+This class is based on the vtkGenericRenderWindowInteractor and is
+therefore fairly powerful.  It should also play nicely with the
+vtk3DWidget code.
+
+Created by Prabhu Ramachandran, May 2002
+Based on David Gobbi's QVTKRenderWidget.py
+
+Changes by Gerard Vermeulen Feb. 2003
+ Win32 support.
+
+Changes by Gerard Vermeulen, May 2003
+ Bug fixes and better integration with the Qt framework.
+
+Changes by Phil Thompson, Nov. 2006
+ Ported to PyQt v4.
+ Added support for wheel events.
+
+Changes by Phil Thompson, Oct. 2007
+ Bug fixes.
+
+Changes by Phil Thompson, Mar. 2008
+ Added cursor support.
+
+Changes by Rodrigo Mologni, Sep. 2013 (Credit to Daniele Esposti)
+ Bug fix to PySide: Converts PyCObject to void pointer.
+
+Changes by Greg Schussman, Aug. 2014
+ The keyPressEvent function now passes keysym instead of None.
+
+Changes by Alex Tsui, Apr. 2015
+ Port from PyQt4 to PyQt5.
+
+Changes by Fabian Wenzel, Jan. 2016
+ Support for Python3
+
+Changes by Tobias Hänel, Sep. 2018
+ Support for PySide2
+
+Changes by Ruben de Bruin, Aug. 2019
+ Fixes to the keyPressEvent function
+
+Changes by Chen Jintao, Aug. 2021
+ Support for PySide6
+"""
+
+# Check whether a specific PyQt implementation was chosen
+try:
+    import vtkmodules.qt
+    PyQtImpl = vtkmodules.qt.PyQtImpl
+except ImportError:
+    pass
```

Comment:
These coverage comments are not really helpful or relevant if we are verbatim copying. We should add a [.coveragerc](https://github.com/mne-tools/mne-python/blob/main/.coveragerc)

Reply:
it's not verbatim copying though. I had to adapt it to support PyQt6

Reply:
Sounds like you need to make a PR to VTK for this, assuming it actually works :)

Reply:
Once I'm happy with my experiments :+1: